File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.found = 0
        self.res=True
        py = "global variables"
    
        def helper(node):
            
            
            if(node==None):  return 0
        
            a, b =helper(node.left), helper(node.right)
        
            if(self.found): 
                return 
        
        
            if(abs(a-b)>=2):
                self.res=0
                self.found=1
                return 
        
       
            return max(a,b)+1
   
        helper(root)
        return self.res

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    let found=false, res=true
    
    function helper(node){
        if(node===null)
            return 0
        
        let a=helper(node.left)
        let b=helper(node.right)
        
        if(found){
            return 
        }
        
        if(Math.abs(a-b)>=2){ 
            res=false
            found=true
            return
        
        }
        return Math.max(a,b)+1
    }
    helper(root)
    return res
};